app/src/components/OrderInformation.tsx

```diff
@@ -0,0 +1,218 @@
+import { MESSAGE_TYPE_ENUM } from '../constants/Enums';
+import {
+    convertNanosecondsUTCToCurrentTimezone, getLocalTimeString,
+    nanosecondsToString,
+    splitNanosecondEpochTimestamp,
+} from '../utils/date-utils';
+import { showOrderInfoDrawer } from '../actions/actions';
+
+const styles = createStyles(Styles);
+
+interface Props extends WithStyles<typeof styles>{
+    orderId : number,
+    quantity: number,
+    lastModified: string,
+    createdOn: string,
+    price: number,
+    messages: Array<Message>,
+    onOrderInfoClosed: Function,
+}
+
+interface State {
+    right: boolean,
+}
+
+class OrderInformation extends Component<Props, State> {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            right: true,
+        };
+    }
+
+    /**
+     * @desc Renders the Drawer slider when open is set to true.
+     * @param side :can be set to 'left' 'right' 'top' 'bottom' setting the side of the screen the drawer is shown
+     * @param open :boolean set to true when drawer is open
+     */
+    renderToggleDrawer(side, open) {
+        const { onOrderInfoClosed } = this.props;
+        return event => {
+            if (event && event.type === 'keydown' && (event.key === 'Tab' || event.key === 'Shift')) {
+                return;
+            }
+            if (!open) {
+                const orderInformationDrawer: OrderInformationDrawer = {
+                    showOrderInfoDrawer: false,
+                };
+                onOrderInfoClosed(orderInformationDrawer);
+            }
+
+            this.setState({ right: true, [side]: open });
+        };
+    }
+
+    /**
+     * @desc Renders the message list table inside the drawer with 3 columns: timestamp, type and quantity.
+     * @returns {*}
+     */
+    renderMessageListTable() {
+        const { messages, classes } = this.props;
+        return (
+            <Table
+                className={classes.messagesTable}
+                id={'orderMessageListTable'}
+            >
+                <TableHead>
+                    <TableRow>
+                        <TableCell>Timestamp</TableCell>
+                        <TableCell>Type</TableCell>
+                        <TableCell>Quantity</TableCell>
+                    </TableRow>
+                </TableHead>
+                <TableBody>
+                    {messages.map(message => {
+                        const {
+                            timestamp, message_type, share_qty, order_id,
+                        } = message;
+
+                        const { timeNanoseconds } = splitNanosecondEpochTimestamp(bigInt(timestamp));
+                        const time = nanosecondsToString(
+                            convertNanosecondsUTCToCurrentTimezone(bigInt(timeNanoseconds)).valueOf(),
+                        );
+
+                        return (
+                            <TableRow
+                                key={order_id}
```

Comment:
Won't this cause an issue since messages will have the same order id ?

Reply:
order ids are unique i believe
@chimano can you confirm?

Reply:
Your keys won't be unique, all messages will have the same order_id as they all relate to the same order. You could use the sod_offset which uniquely identifies each message.

Reply:
Perfect thanks Lucas, I'll do that then! Good catch Chris!